### PR TITLE
[SPARK-14412][ML][WIP]spark.ml ALS prefered storage level Params

### PR DESCRIPTION
**Pull request by rishabhbhardwaj**
## What changes were proposed in this pull request?

Added storage level params in ALSParams.
## How was this patch tested?

Passing existing Unit test Cases

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala [hunk 1]
```diff
@@ -169,6 +169,41 @@ private[recommendation] trait ALSParams extends ALSModelParams with HasMaxIter w
...
+
+  /**
+    * :: DeveloperApi ::
+    * Sets storage level for intermediate RDDs (user/product in/out links). The default value is
+    * `MEMORY_AND_DISK`. Users can change it to a serialized storage, e.g., `MEMORY_AND_DISK_SER`
+    * and set `spark.rdd.compress` to `true` to reduce the space requirement, at the cost of speed.
+    * @group expertParam
+    */
+  @DeveloperApi
+  @Since("2.0.0")
```
Comment (MLnick) [inline mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala]: Please be consistent with the approach used for other params in `ALSParams` trait. Params work differently for `ml` compared to `mllib`.

**Comment by MLnick** [comment]
You can add the new params (with non-default settings) to `allEstimatorParamSettings` in the `ml.recommendation.ALSSuite` tests (they will then be tested under the "test read/write" case).

File: mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala (hunk 1 above)
Comment (MLnick) [inline mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala]: @jkbradley thoughts?

Comment (yanboliang) [inline mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala]: I think these should be standard Params whose type is ```Param[String]```, then it's also easy to be exposed at Python API. Meanwhile, we should check users' input is in supported ```StorageLevel```s.

Comment (MLnick) [inline mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala]: I agree. @rishabhbhardwaj do you have time to work on this PR to get it in line with ML param style? If not I can do it.

**Comment by rishabhbhardwaj** [comment]
@MLnick : I need to study how MLParam works.For now,I am closing this PR.You can take this forward.Thanks.

File: mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala (hunk 1 above)
Comment (jkbradley) [inline mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala]: Sorry for the slow response!  Yes, +1 for using standard Params, and String seems like a reasonable choice.

**Comment by jkbradley** [comment]
@rishabhbhardwaj To study that, I'd recommend picking an algorithm + a Param in it, and finding all usages of it, the Param set/get methods, and tests for it.  That should give you a complete example to follow.

**Comment by MLnick** [comment]
@rishabhbhardwaj I opened #12660 to address this. You can take a look at it to get an idea of how the params work and the test cases for this which are a little more complex than the usual "add a param to ML" scenario.

**Comment by rishabhbhardwaj** [comment]
@MLnick @jkbradley Thanks, I am on it.I tried making the changes as per your comments but closing this PR now, as nick has created a PR for this.

**Pull request closed without merging**